### ISSUE TITLE
import not working

### ISSUE THREAD

**Issue by Muddster89** [evievsbrain/ms08_067_walkthrough#1]
15 from threading import Thread 
16 import sys 
 
18 try: 
19     from impacket import smb 
20     from impacket import uuid 
21     from impacket.dcerpc import dcerpc 
22     from impacket.dcerpc.v5 import transport 
23     from impacket.structure import Structure 
24     from impacket.dcerpc import ndrutils 
25     from impacket.dcerpc.v5 import ndr 
26 except ImportError, _: 
27     print 'Install the following library to make this script work' 
28     print 'Impacket : http://oss.coresecurity.com/projects/impacket.html' 
29     print 'PyCrypto : http://www.amk.ca/python/code/crypto.html' 
30     sys.exit(1) 

First I want to say awesome writeup on the exploit. 

I am currently having an issue using the exploit as everytime I try to ./xxxx.py xx.xx.xx.xx (1/2), I get your above message for the import error. I have done a gitpull and install of the impacket.html from their GitHub repo and done the install about 4 times just to be sure I didn't screw it up. The website linked here in the code doesn't exsist anymore:

https://pypi.python.org/pypi/pycrypto/2.0.1

Is the only thing that I can find relating to it.

Second, do I need to run the exploit from within the gitpull? or should it be able to be run from anywhere? I assume anywhere just like other scripts but wasn't sure if you were including anything in the repo ( I couldn't find anything that showed you were, like a lib file or something)

Thanks!

**Comment by evievsbrain** [comment]
Hi, I know it's been almost 2 years since you asked this! I apologise, I haven't actively maintained my Github account, but I just wanted to finally address this observation.

The unfortunate thing about relying on other peoples' code (as I have done in this tutorial) is that the internet changes and links die. What I didn't have time to do, and evidently should have done, is to reverse engineer exactly what impacket was doing to interact with the DCERPC and SMB services and documented that too. Unfortunately I didn't have the skills at the time to do this, or the foresight to save this particular version of impacket library on my own infrastructure.

In short, it looks like in order to get this working again reliably we'd have to dig all the way down to implement the missing functionality ourselves, or find another low level library to replace the functionality.

I hope the guide as it is helped you to save some time in your initial research into SMB, and I'll make sure to clearly note the issues in the writeup.
